### ISSUE TITLE
Plugin is unusable on fancy browser editors like HackerRank's vim mode due to escape key shenanigans

### ISSUE THREAD

**Issue by EnacheB** [ueokande/vim-vixen#478]
### I'm opening this issue because:

- [ ] I'll report a bug
- [ ] I'll propose a new feature

### Description
Vim Vixen takes control of your "escape" key even when toggled off. Try vim mode on hackerrank, for example, you'll note that you can't escape back into normal mode, instead you will de-focus the text field. Maybe a custom binding for the escape key can fix this?

#### Steps to Reproduce

1. Open any problem on hackerrank
2. set your HR editor mode to "vim"
3. Type some code then hit escape to enter normal mode. You won't be in the text box anymore
4. Do the above again with vim-vixen toggled off. You will have the same issue
5. Disable vim vixen in firefox's settings. Now you can easily switch between modes on HR

#### System configuration

- Operating system: Arch Linux
- Firefox version: 62.0.2
- Vim-Vixen version: 0.18

**Comment by sukima** [comment]
I can confirm this has issues in many editors (code-mirror, ace, etc.). The only workaround I've found is to use <kbd>CTRL-C</kbd>. A hassle but at least it stops the mess you get yourself into when you press <kbd>Esc</kbd>.

**Comment by xeruf** [comment]
Yeah this is really awkward. At least disabling the plugin on-site should fix it, but it is only fixed by disabling it completely.
https://github.com/codemirror/CodeMirror/issues/5038
